### PR TITLE
Format init accessor on typing '}'

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/FormattingRangeHelper.cs [hunk 1]
```diff
@@ -368,6 +368,7 @@ private static bool IsSpecialContainingNode(SyntaxNode node)
                 node.Kind() == SyntaxKind.FixedStatement ||
                 node.Kind() == SyntaxKind.GetAccessorDeclaration ||
                 node.Kind() == SyntaxKind.SetAccessorDeclaration ||
+                node.Kind() == SyntaxKind.InitAccessorDeclaration ||
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/FormattingRangeHelper.cs:371]: 📝 This method appears to be related to the automatic formatting that occurs on `}` or `;`. We should add a test which fails prior to this change, but succeeds after. Perhaps something like this?

```
class C
{
  int Value
  {
    get;
init; // <- verify that typing this ';' triggers indentation
  }
}
```


```
class C
{
  int Value
  {
    get;
init
{
} // <- verify that typing this '}' triggers indentation of all three lines
  }
}
```

Comment (Youssef1313) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/FormattingRangeHelper.cs:371]: @sharwell I looked at the current tests, but the way they're written makes it a bit hard for me to test this scenario.
Would be glad if you can take a look:

https://github.com/Youssef1313/roslyn/blob/patch-39/src/Workspaces/CoreTest/UtilityTest/FormattingRangeHelperTests.cs

File: src/EditorFeatures/CSharpTest/Formatting/Indentation/SmartTokenFormatterFormatRangeTests.cs [hunk 2]
```diff
@@ -570,29 +570,32 @@ int Prop
         [WpfFact]
```
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/Formatting/Indentation/SmartTokenFormatterFormatRangeTests.cs:570]:
```suggestion
        [WpfTheory]
```

**Review by sharwell — APPROVED** [review]

**Comment by Youssef1313** [comment]
@sharwell Is this ready to merge?

**Pull request merged**